File: js/static/raml-language-server/src/common/logger.ts
export enum MessageSeverity {
    DEBUG_DETAIL = 0,
    DEBUG = 1,
    DEBUG_OVERVIEW = 2,
    WARNING = 3,
    ERROR = 4
}

/**
 * Settings of logger to filter the msssages.
 */
export interface ILoggerSettings {

    /**
     * If true, disables all logging.
     */
    disabled?: boolean;

    /**
     * List of components, which are allowed to appear in log.
     * If empty or null, all components are allowed (except those excplicitly denied).
     */
    allowedComponents?: string[];

    /**
     * Components, which never appear in the log
     */
    deniedComponents?: string[];

    /**
     * Messages with lower severity will not appear in log.
     */
    maxSeverity?: MessageSeverity;

    /**
     * Messages having more length will be cut off to this number.
     */
    maxMessageLength?: number;
}

export interface ILogger {
    /**
     * Logs a message
     * @param message - message text
     * @param severity - message severity
     * @param component - component name
     * @param subcomponent - sub-component name
     */
    log(message: string, severity: MessageSeverity,
        component?: string, subcomponent?: string): void;

    /**
     * Logs a DEBUG severity message.
     * @param message - message text
     * @param component - component name
     * @param subcomponent - sub-component name
     */
    debug(message: string,
          component?: string, subcomponent?: string): void;

    /**
     * Logs a DEBUG_DETAIL severity message.
     * @param message - message text
     * @param component - component name
     * @param subcomponent - sub-component name
     */
    debugDetail(message: string,
                component?: string, subcomponent?: string): void;

    /**
     * Logs a DEBUG_OVERVIEW severity message.
     * @param message - message text
     * @param component - component name
     * @param subcomponent - sub-component name
     */
    debugOverview(message: string,
                  component?: string, subcomponent?: string): void;

    /**
     * Logs a WARNING severity message.
     * @param message - message text
     * @param component - component name
     * @param subcomponent - sub-component name
     */
    warning(message: string,
            component?: string, subcomponent?: string): void;

    /**
     * Logs an ERROR severity message.
     * @param message - message text
     * @param component - component name
     * @param subcomponent - sub-component name
     */
    error(message: string,
          component?: string, subcomponent?: string): void;

    /**
     * Sets logger configuration, both for the server and for the client.
     * @param loggerSettings
     */
    setLoggerConfiguration(loggerSettings: ILoggerSettings);
}

File: als-server/shared/src/main/scala/org/mulesoft/language/common/logger/Logger.scala
package org.mulesoft.language.common.logger

import org.mulesoft.language.common.logger.MessageSeverity.MessageSeverity

trait Logger {

  /**
    * Logs a message
    *
    * @param message      - message text
    * @param severity     - message severity
    * @param component    - component name
    * @param subComponent - sub-component name
    */
  def log(message: String, severity: MessageSeverity, component: String, subComponent: String): Unit

  /**
    * Logs a DEBUG severity message.
    *
    * @param message      - message text
    * @param component    - component name
    * @param subComponent - sub-component name
    */
  def debug(message: String, component: String, subComponent: String): Unit

  /**
    * Logs a DEBUG_DETAIL severity message.
    *
    * @param message      - message text
    * @param component    - component name
    * @param subComponent - sub-component name
    */
  def debugDetail(message: String, component: String, subComponent: String): Unit

  /**
    * Logs a DEBUG_OVERVIEW severity message.
    *
    * @param message      - message text
    * @param component    - component name
    * @param subComponent - sub-component name
    */
  def debugOverview(message: String, component: String, subComponent: String): Unit

  /**
    * Logs a WARNING severity message.
    *
    * @param message      - message text
    * @param component    - component name
    * @param subComponent - sub-component name
    */
  def warning(message: String, component: String, subComponent: String): Unit

  /**
    * Logs an ERROR severity message.
    *
    * @param message      - message text
    * @param component    - component name
    * @param subComponent - sub-component name
    */
  def error(message: String, component: String, subComponent: String): Unit

  /**
    * Sets logger configuration, both for the server and for the client.
    *
    * @param settings - logger settings object
    */
  def withSettings(settings: LoggerSettings): this.type
}
